src/Workspaces/Core/Portable/Workspace/Host/Metadata/IAnalyzerAssemblyLoaderProvider.cs

```diff
@@ -25,7 +25,13 @@ internal interface IAnalyzerAssemblyLoaderProvider : IWorkspaceService
     /// Creates a fresh shadow copying loader that will load all <see cref="AnalyzerReference"/>s and <see
     /// cref="ISourceGenerator"/>s in a fresh <see cref="AssemblyLoadContext"/>.
     /// </summary>
-    IAnalyzerAssemblyLoaderInternal CreateNewShadowCopyLoader();
+    IAnalyzerAssemblyLoaderInternal CreateNewShadowCopyLoader() => CreateNewShadowCopyLoader(collectOnDispose: false);
```

Comment:
this seems undesirable for the workspace layer.  can we make this either `collectOnDispose: true`, or have this not be optional so that we are forced to decide what to do with every caller.